numpy/lib/financial.py

```diff
@@ -208,12 +215,15 @@ def pmt(rate, nper, pv, fv=0, when='end'):
     """
     when = _convert_when(when)
     (rate, nper, pv, fv, when) = map(np.asarray, [rate, nper, pv, fv, when])
-    temp = (1 + rate)**nper
-    mask = (rate == 0.0)
-    np.copyto(rate, 1.0, where=mask)
-    z = np.zeros(np.broadcast(rate, nper, pv, fv, when).shape)
-    fact = np.where(mask != z, nper + z, (1 + rate*when)*(temp - 1)/rate + z)
-    return -(fv + pv*temp) / fact
+
+    # use different payment if rate == 0
+    zero_rate_mask = rate == 0
+    compounded_rate = (1 + rate)**nper
+    payment = np.asarray(np.divide(-(fv + pv*compounded_rate)*rate,
+                                   (1 + rate*when)*(compounded_rate - 1),
+                                   where=~zero_rate_mask))
+    np.copyto(payment, -(fv + pv)/nper, where=zero_rate_mask)
```

Comment:
(Sorry, got interrupted half-way through reviewing :-))
I'd put spaces around most or all of these binary operators (e.g. `(1 + rate * when) * (compounded_rate - 1)`), though in this case it's more a matter of personal taste than rigid policy. (In general spaces around operators is PEP8's recommendation, but there's an escape clause for higher-priority operators that could apply here if one wanted.)